src/test/java/org/kohsuke/github/GHRepositoryTest.java

```diff
@@ -249,6 +249,27 @@ public void testUpdateRepository() throws Exception {
         assertThat(redux.getDescription(), equalTo(updatedDescription));
     }
 
+    @Test
+    public void testGetRepositoryWithVisibility() throws IOException {
```

Comment:
Definitely want to test the `INTERNAL` value as well.  If you can create a test for it and then manually modify the data , that would be great.  After recording, have the test use ` snapshotNotAllowed();` so no one accidentally deletes that data.